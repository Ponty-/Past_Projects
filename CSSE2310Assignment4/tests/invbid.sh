#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port .. 1.deck &
spid=$!
$bindir/client499 A game $port > $res/$name.a.out 2>$res/$name.a.err < iamoves &
pid1=$!
$bindir/client499 B game $port > $res/$name.b.out 2>$res/$name.b.err < ibmoves &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.out 2>$res/$name.c.err < icmoves &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.out 2>$res/$name.d.err < idmoves &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null
kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.out $right/$name.$v.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.out $right/$name.$v.out"
   exit 2
fi

diff -q $res/$name.$v.err $right/$name.$v.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.err $right/$name.$v.err"
   exit 3
fi

done

echo OK
exit 0

