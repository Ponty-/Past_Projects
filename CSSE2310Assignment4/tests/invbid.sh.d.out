Info: ..
Info: Team1: A, C
Info: Team2: B, D
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Info: A bids 7D
Info: B passes
Info: C bids 9H
Info: C plays AD
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[D] play> Info: A plays KD
Info: B plays 3D
Info: C won
Info: C plays QD
S: K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[D] play> Info: A plays TD
Info: B plays 6D
Info: C won
Info: C plays 8D
S: J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[D] play> Info: A plays 5D
Info: B plays 9D
Info: B won
Info: B plays 6S
Info: C plays 9S
S: 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[S] play> Info: A plays 7S
Info: C won
Info: C plays KC
S: 5 3 2
C: J 8 5
D:
H: Q 7 3
[C] play> Info: A plays 3C
Info: B plays 2C
Info: C won
Info: C plays JH
S: 5 3 2
C: 8 5
D:
H: Q 7 3
[H] play> Info: A plays 2H
Info: B plays 4H
Info: C won
Info: C plays 4S
S: 5 3 2
C: 8 5
D:
H: Q 7
[S] play> Info: A plays QS
Info: B plays TS
Info: A won
Info: A plays 7D
Info: B plays JD
Info: C plays 4D
S: 5 2
C: 8 5
D:
H: Q 7
[D] play> Info: B won
Info: B plays AC
Info: C plays 9C
S: 2
C: 8 5
D:
H: Q 7
[C] play> Info: A plays 7C
Info: B won
Info: B plays TC
Info: C plays 4C
S: 2
C: 5
D:
H: Q 7
[C] play> Info: A plays QC
Info: A won
Info: A plays 6H
Info: B plays 9H
Info: C plays 8H
S: 2
C:
D:
H: Q 7
[H] play> Info: B won
Info: B plays 6C
Info: C plays 2D
S: 2
C:
D:
H: Q
[C] play> Info: A plays AH
Info: A won
Info: A plays TH
Info: B plays KH
Info: C plays 5H
S:
C:
D:
H: Q
[H] play> Info: B won
Info: Team 1=-300, Team 2=0
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Info: A bids 9H
Info: A plays KD
Info: B plays 3D
Info: C plays AD
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[D] play> Info: C won
Info: C plays JH
S: A K J 8 5 3
C: J 8 5
D:
H: Q 7 3
[H] play> Info: A plays 2H
Info: B plays 4H
Info: D won
S: A K J 8 5 3
C: J 8 5
D:
H: 7 3
Lead> Info: A plays QS
Info: B plays TS
Info: C plays 9S
Info: D won
S: K J 8 5 3
C: J 8 5
D:
H: 7 3
Lead> Info: A plays 7S
Info: B plays 6S
Info: C plays 4S
Info: D won
S: J 8 5 3
C: J 8 5
D:
H: 7 3
Lead> Info: A plays QC
Info: B plays AC
Info: C plays KC
Info: D won
S: 8 5 3
C: J 8 5
D:
H: 7 3
Lead> Info: A plays 3C
Info: B plays TC
Info: C plays 9C
Info: D won
S: 5 3
C: J 8 5
D:
H: 7 3
Lead> Info: A plays 7C
Info: B plays 6C
Info: C plays 4C
Info: D won
S: 3
C: J 8 5
D:
H: 7 3
Lead> Info: A plays 5D
Info: B plays 2C
Info: C plays QD
Info: D won
S:
C: J 8 5
D:
H: 7 3
Lead> Info: A plays TD
Info: B plays JD
Info: C plays 8D
Info: D won
S:
C: 8 5
D:
H: 7 3
Lead> Info: A plays 7D
Info: B plays 6D
Info: C plays 4D
Info: D won
S:
C: 5
D:
H: 7 3
Lead> Info: A plays AH
Info: B plays 9D
Info: C plays 2D
Info: A won
Info: A plays 6H
Info: B plays KH
Info: C plays 8H
S:
C:
D:
H: 7 3
[H] play> Info: B won
Info: B plays 9H
Info: C plays 5H
S:
C:
D:
H: 3
[H] play> Info: A plays TH
Info: A won
Info: Team 1=-600, Team 2=0
Info: Winner is Team 2
